tests/langchain/test_langchain_autolog.py

```diff
@@ -770,284 +781,270 @@ async def on_chain_end(self, outputs: Dict[str, Any], **kwargs: Any) -> None:
+    assert handlers == original_handlers
+
+    # The original callback is called by the chain
+    if handlers and invoke_arg:
+        handlers[0].logs = ["chain_start", "chain_end"]
 
 
 @pytest.mark.parametrize("invoke_arg", ["args", "kwargs", None])
-@pytest.mark.parametrize(
-    "generate_callbacks",
-    [
-        lambda: [CustomCallbackHandler()],
-        lambda: BaseCallbackManager([CustomCallbackHandler()]),
-        lambda: None,
-        lambda: [AsyncCustomCallbackHandler()],
-        lambda: BaseCallbackManager([AsyncCustomCallbackHandler()]),
-    ],
-)
+@pytest.mark.parametrize("config", _CONFIG_PATTERNS + _ASYNC_CONFIG_PATTERNS)
 @pytest.mark.asyncio
-async def test_langchain_autolog_callback_injection_in_ainvoke(invoke_arg, generate_callbacks):
-    mlflow.langchain.autolog(
-        log_models=True,
-        extra_tags={"test_tag": "test"},
-        log_inputs_outputs=True,
-    )
-    callbacks = generate_callbacks()
+async def test_langchain_autolog_callback_injection_in_ainvoke(invoke_arg, config):
+    mlflow.langchain.autolog()
 
-    with mlflow.start_run() as run, _mock_openai_arequest():
-        model = create_openai_llmchain()
+    model = create_openai_runnable()
+    original_handlers = _extract_callback_handlers(config)
+
+    with _mock_openai_arequest():
+        input = {"product": "MLflow"}
         if invoke_arg == "args":
-            await model.ainvoke("MLflow", RunnableConfig(callbacks=callbacks))
+            await model.ainvoke(input, config)
         elif invoke_arg == "kwargs":
-            await model.ainvoke("MLflow", config=RunnableConfig(callbacks=callbacks))
+            await model.ainvoke(input, config=config)
         elif invoke_arg is None:
-            await model.ainvoke("MLflow")
-        assert mlflow.active_run() is not None
-    run_data = MlflowClient().get_run(run.info.run_id).data
-    assert run_data.tags["test_tag"] == "test"
-    assert run_data.tags["mlflow.autologging"] == "langchain"
-    # original callback still works as expected
-    if invoke_arg and callbacks:
-        if isinstance(callbacks, BaseCallbackManager):
-            assert callbacks.handlers[0].logs == ["chain_start", "chain_end"]
-        else:
-            assert callbacks[0].logs == ["chain_start", "chain_end"]
+            await model.ainvoke(input)
+
+    ## Trace doesn't work for .abatch() patching
+    # traces = get_traces()
+    # assert len(traces) == 1
+    # assert traces[0].info.status == "OK"
+    # assert traces[0].data.spans[0].name == "RunnableSequence"
+    # assert traces[0].data.spans[0].inputs == input
+    # assert traces[0].data.spans[0].outputs == TEST_CONTENT
+
+    # Original callback should not be mutated
+    handlers = _extract_callback_handlers(config)
+    assert handlers == original_handlers
+
+    # The original callback is called by the chain
+    if handlers and invoke_arg:
+        handlers[0].logs = ["chain_start", "chain_end"]
```

Comment:
Is this meant to assert?

Reply:
ahh true, thanks for catching!

Reply:
oh faced weird bug in LangChain... somehow a callback is called different times when passed via list or callback manager... will see how I can handle it

Reply:
Seems like an upstream bug to me. If a list is supplied, can we just wrap in a manager instance?

Reply:
Yea pretty much... I filed a bug report: https://github.com/langchain-ai/langchain/issues/24642

>  If a list is supplied, can we just wrap in a manager instance?

I keep the current behavior and just adjust the assertion to avoid regression, but we may need to do that depending on their response:(